pkg/agent/agent.go

```diff
@@ -262,11 +269,76 @@ func (s *NimbessAgent) updatePipelinesEgress(port *network.EgressPort, excludedK
 	return nil
 }
 
+// removeEgressPort removes an Egress Port from a Meta Pipeline by name
+func (s *NimbessAgent) removeEgressPort(name string, metaKey string) error {
+	log.Debugf("Removing Egress Port: %s", name)
+	if len(s.MetaPipelines[metaKey].EgressPorts) == 1 {
+		if s.MetaPipelines[metaKey].EgressPorts[0].GetName() == name {
```

Comment:
sort of related... down below the code appends to egressPorts so it seems the list is more than 1. If so and if this check for size 1 is correct, then it would be good to add a comment to describe the specific condition here.

Reply:
I'm checking that the slice has only 1 element, because if that is the case the iteration code beneath it will fail to pop the element out of the slice (out of bounds error). Cannot find a built in way to pop a single element from a slice so using a method I found in stackoverflow:
https://stackoverflow.com/questions/37334119/how-to-delete-an-element-from-a-slice-in-golang/37335777#37335777

append doesn't mean the slice is more than 1, it just means a list exists. The slice could still be empty.

Reply:
Add something like this as a comment, otherwise we will get this question again about why a single element is checked